resilience4j-retry/src/main/java/io/github/resilience4j/retry/Retry.java

```diff
@@ -119,6 +119,37 @@ static Retry ofDefaults(String name) {
 		};
 	}
 
+	/**
+	 * Returns a Supplier of type Future which is decorated by Retry.
+	 *
+	 * Any delay which is specified in the retry context is accomplished by calling `Thread.sleep()`.  Thus it is not
+	 * recommended that `executor` be a single-threaded or otherwise fixed-thread-pool executor.
+	 *
+	 * @param retry    the retry context
+	 * @param executor execution service to use to pull and wait for supplied Futures
+	 * @param supplier future supplier
+	 * @param <T>      type of future result
+	 * @return decorated supplier
+	 */
+	static <T> Supplier<Future<T>> decorateFuture(Retry retry, ExecutorService executor, Supplier<Future<T>> supplier) {
```

Comment:
Please have a look at `Retry.decorateCompletionStage` and `Retry.AsyncRetryBlock`.
I think we should use a ScheduledExecutorService and schedule a retry attempt. This sync version blocks with a `Thread.sleep`.